framework/web/CHttpRequest.php

```diff
@@ -1364,6 +1365,12 @@ public function validateCsrfToken($event)
 					$maskedUserToken=$this->getDelete($this->csrfTokenName);
 			}
 
+            // @see https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#jquery
+            // Assumes header is named like Yii-Csrf-Token if csrfTokenName is YII_CSRF_TOKEN
+            if (empty($maskedUserToken)) {
+                $maskedUserToken = $_SERVER['HTTP_' . $this->csrfTokenName];
```

Comment:
Woudn't it error here if there's no header?

Reply:
Yep, will fix.